### PR TITLE
Split the mutex for process and callback update

### PR DESCRIPTION
**Pull request by reichlfl**
Update of the process_list_ and process_list_update_listener_ was using the same mutex.

This causes a deadlock when the process_list_update_listener_ calls GetProcessList() when invoked from the WorkerFunction().
Problem could be observed in the new UI beta branch: Process List waits to be loaded indefinitely after connecting to an instance.

### CODE REVIEW

**Review by antonrohr — APPROVED** [review]
You solve my nitpicks faster than I can comment on them :) 
LGTM, thank you a lot!

File: OrbitClientServices/ProcessManager.cpp
Comment (dimitry-) [inline OrbitClientServices/ProcessManager.cpp:138]: I am guessing the problem here is that listener may call GetProcessList which will lead to recursive lock. Have you considered passing reference to process_list_ here maybe?

**Comment by dimitry-** [comment]
> Update of the process_list_ and process_list_update_listener_ was using the same mutex.
> 
> This causes a deadlock when the process_list_update_listener_ calls GetProcessList() when invoked from the WorkerFunction().
> Problem could be observed in the new UI beta branch: Process List waits to be loaded indefinitely after connecting to an instance.

also - this should go to commit message probably. I did not read this because I was looking at the commit message :)

File: OrbitClientServices/ProcessManager.cpp
Comment (antonrohr) [reply OrbitClientServices/ProcessManager.cpp:138]: I like this Idea as well. Then I would get rid of the function GetProcessList() altogether, which requires a small change in App.cpp where GetProcessList() is used. 
I also dont mind how the change is right now.

Comment (reichlfl) [reply OrbitClientServices/ProcessManager.cpp:138]: Good point. I changed that, PTAL. Nevertheless, I think it's a good idea to keep this as two mutexes, since they guard different objects, and it seems excessive to block the whole process list update during the callback when the callback operates on a copy anyways.

**Comment by reichlfl** [comment]
> > Update of the process_list_ and process_list_update_listener_ was using the same mutex.
> > This causes a deadlock when the process_list_update_listener_ calls GetProcessList() when invoked from the WorkerFunction().
> > Problem could be observed in the new UI beta branch: Process List waits to be loaded indefinitely after connecting to an instance.
> 
> also - this should go to commit message probably. I did not read this because I was looking at the commit message :)

True, will do that with the final rebase.

File: OrbitClientServices/include/OrbitClientServices/ProcessManager.h
Comment (dimitry-) [inline OrbitClientServices/include/OrbitClientServices/ProcessManager.h:38]: maybe keep vector in definitions? this using is hiding information about the type... I almost confused it with class we used to have in OrbitCore.

File: OrbitClientServices/ProcessManager.cpp
Comment (dimitry-) [inline OrbitClientServices/ProcessManager.cpp:132]: Does anyone use process_list_ ?

File: OrbitClientServices/include/OrbitClientServices/ProcessManager.h
Comment (reichlfl) [reply OrbitClientServices/include/OrbitClientServices/ProcessManager.h:38]: I think it's more readable, and there's IDEs to display type info ;-) But in all seriousness, I don't have a strong opinion, I wasn't touching this part of the code so much before, so if it's confusing I'll keep the std::vector<> in the definitions.

File: OrbitClientServices/ProcessManager.cpp
Comment (reichlfl) [reply OrbitClientServices/ProcessManager.cpp:132]: Hm, good point. Actually, no. It's returned by GetProcessList(), but after this change, this isn't used at all. But I'm not sure if it will be needed with @antonrohr 's changes?

The current fix is in his change as a cherrypick because it was causing a crash, so I don't want to change this all too much before rebasing, but we could also not merge this fix and remove the whole GetProcessList() (and probably the mutex altogether...) afterwards.

**Comment by reichlfl** [comment]
Handled with https://github.com/google/orbit/pull/1574, closing

**Pull request closed without merging**
